exonum-java-binding-fakes/src/main/java/com/exonum/binding/fakes/mocks/MockInteraction.java

```diff
@@ -0,0 +1,90 @@
+package com.exonum.binding.fakes.mocks;
+
+import com.google.gson.Gson;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+/**
+ * Helper class to make testing callbacks from native code easier. It allows simple tracking of
+ * interaction with a specific method of a mocked object and then return it (by calling
+ * getInteractions() method from native side) as a JSON string.
+ */
+public class MockInteraction {
+  private final String[] arguments;
+  private final List<String> interactions = new ArrayList<>();
+
+  private MockInteraction(String[] arguments) {
+    this.arguments = arguments;
+  }
+
+  /**
+   * Creates new instance of {@link MockInteraction} configured with expected names of arguments for
+   * a mocked method.
+   *
+   * @param expectedArguments names of expected arguments for a mocked method in their right order
+   * @return New and configured instance of {@link MockInteraction}
+   */
+  public static MockInteraction createInteraction(String[] expectedArguments) {
```

Comment:
Is there a specific reason to use static factory method?

Reply:
No, just wanted to provide a user with straightforward method that creates required object. Perhaps, within the current implementation it's not important.